engine/runtime/src/main/java/org/enso/interpreter/instrument/job/VisualizationResult.java

```diff
@@ -10,6 +11,38 @@ public final class VisualizationResult {
   private VisualizationResult() {
   }
 
+  /**
+   * Extracts a string representation for a polyglot exception.
+   *
+   * @param exexception the exception
+   * @return message representing the exception
+   */
+  @CompilerDirectives.TruffleBoundary
+  public static String findExceptionMessage(Throwable ex) {
+    Object msg = ex;
+    var iop = InteropLibrary.getUncached();
+    if (iop.hasExceptionMessage(ex)) {
+      try {
+        msg = iop.getExceptionMessage(ex);
+      } catch (UnsupportedMessageException ignore) {
+        throw CompilerDirectives.shouldNotReachHere(ignore);
```

Comment:
I think we typically throw illegal state exception. is  `CompilerDirectives.shouldNotReachHere` the new recommended way?

Reply:
I believe `shouldNotReachHere` has a special treatment in the compiler and leads to shorter code then `new IllegalStateException`. Because in case of `IllegalStateException` the constructor is still executed in partial evaluation mode - e.g. the generated code is bigger than in case of `shouldNotReachHere`.